### ISSUE TITLE
[Bug]: Can't move HQ assets sometimes

### ISSUE THREAD

**Issue by doommetal** [official-antistasi-community/A3-Antistasi#1350]
**Describe the bug**
Can't move HQ assets. Error says it's because I already have something attached even though I don't.

**To Reproduce**
Couldn't figure out exactly why/when it happens.
Håkon on discord: arsenal sometimes attaches objnull to you

**Expected behaviour**
Expected to be able to move the HQ asset.

**Screenshots**
![Image of error msg](https://cdn.discordapp.com/attachments/644893868780683304/737460314781646888/unknown.png)

**Environment (please complete the following information):**
Version: 2.3
Map (specify if Altis Blufor): Kunduz

[ ] LAN Hosted
[ ] Private dedicated server
[ ] Private dedicated server with Headless client(s)
[ ] Community Server 1
[x] Community Server 2
[ ] Community Test Server
[ ] Community Test Server 2

Modset: Official server RHS
    
Time the bug occurred (Please leave in GMT+0): Multiple times between 21:00, July 27 and 01:00 July 28

**Comment by doommetal** [comment]
Note to self (or anyone else intending to throw science™ at this):
Try running `attachedObjects player` in debug console when this happens.

**Comment by doommetal** [comment]
Apparently this is caused by other stuff getting attached to the player and not getting removed.
A user on discord reported this result after running `attachedObjects player`:
`[1766b100100# 2219326: helipadempty_f.p3d,<NULL-object>,<NULL-object>]`

Temporary fix to remove the unwanted attached objects:
`{detach _x} forEach attachedObjects player;`

**Comment by doommetal** [comment]
More context:
DoomMetalToday at 9:32 AM
now, why would players randomly get a helipad attached to them? :thinking:
WurzelToday at 9:33 AM
Only function that does that would be the landing scripts for QRFs or attacks, they should attach an empty helipad to the helicopter, how this happens to the players is a mistery for me
DoomMetalToday at 9:33 AM
hm, maybe that thing is leaking a bit then
WurzelToday at 9:34 AM
Might be, easiest way would be to search for the name of empty helipads through the code and check where these are created and attached
DoomMetalToday at 9:34 AM
the guy also had a couple of NULL-objects attached, but i think the helipad would be the one causing the issue here

**Comment by HakonRydland** [comment]
during test server 1 session, problem occurred, during observed attachedObjects was always nullObj. this lead wurzel and I to speculate that it is a issue with it not returning as objNull but nil sometimes, test to see if this fixes it would be to change ln 12 in moveHQobjects from `if ({!(isNull _x)} count (attachedObjects _playerX) != 0) exitWith {["Move HQ", "You have other things attached, you cannot move this"] call A3A_fnc_customHint;};
` to `if ({!(_x isEqualTo objNull)} count (attachedObjects _playerX) != 0) exitWith {["Move HQ", "You have other things attached, you cannot move this"] call A3A_fnc_customHint;};
`

**Comment by jaj22** [comment]
SQF tends to silently skip nils in loops and conditionals:

```
_array = [objNull, nil, objNull];
count _array;                       // returns 3
{ isNil "_x" } count _array;        // returns 1
{ isNull _x } count _array;         // returns 2
{ !(isNull _x) } count _array;      // returns 0
```

I don't see how `isEqualTo objNull` would help. As far as I know it's identical to `isNull` except in the case of other nulls (eg grpNull), and a nil is not a null. The current code should only cause a problem if there are non-null objects (helipads, wat?) attached to the player. I think we'd need to figure out where those are coming from.

**Comment by HakonRydland** [comment]
the only thing that i could see that attaches helipads to the players is the snow script which is currently disabled, however as for the all object null's returning a count thats not 1 or somehow failing it could be a hidden "arma feature" thats undocumented, all i know is that that my code for picking up stuff that is also ignoring object null's was working fine when the issue appeared. however it could also be that count is fucking up as im not using count but select and checking that its an empty array.

**Comment by jaj22** [comment]
Should be finally fixed in #3210
